Makefile

```diff
@@ -2,58 +2,153 @@
-	pip install -r trulens_eval/trulens_eval/requirements.optional.txt
-	pip install -r requirements.docs.txt
-	pip install -r requirements.dev.txt
-
-# Run the code formatter.
-format: .conda/docs
-	$(CONDA) .conda/docs
-	bash format.sh
+REPO_ROOT := $(shell dirname $(realpath $(firstword $(MAKEFILE_LIST))))
+POETRY_DIRS := $(shell find . -not -path "./dist/*" -not -path "./src/dashboard/*" -maxdepth 4 -name "*pyproject.toml" -exec dirname {} \;)
+
+# Create the poetry env for building website, docs, formatting, etc.
+env:
+	poetry install
+
+env-%:
+	poetry install --with $*
+
+env-required:
+	poetry install --only required,tests --sync
+
+env-optional:
+	poetry install --with tests,tests-optional --sync --verbose
+
+
+# Lock the poetry dependencies for all the subprojects.
+lock: $(POETRY_DIRS) clean-dashboard
+	for dir in $(POETRY_DIRS); do \
+		echo "Creating lockfile for $$dir/pyproject.toml"; \
+		poetry lock -C $$dir; \
+	done
+
+# Run the ruff linter.
+lint: env
+	poetry run ruff check --fix
+
+# Run the ruff formatter.
+format: env
+	poetry run ruff format
+
+precommit-hooks:
+	poetry run pre-commit install
+
+run-precommit:
+	poetry run pre-commit run --all-files --show-diff-on-failure
 
 # Start a jupyer lab instance.
-lab:
-	$(CONDA) .conda/docs
-	jupyter lab --ip=0.0.0.0 --no-browser --ServerApp.token=deadbeef
+lab: env
+	poetry run jupyter lab --ip=0.0.0.0 --no-browser --ServerApp.token=deadbeef
+
+# Build the documentation website.
+docs: env-docs $(shell find docs -type f) mkdocs.yml
+	poetry run mkdocs build --clean
+	rm -Rf site/overrides
 
 # Serve the documentation website.
-serve: .conda/docs
-	$(CONDA) .conda/docs
-	mkdocs serve -a 127.0.0.1:8000
+docs-serve: env-docs
+	poetry run mkdocs serve -a 127.0.0.1:8000
 
 # Serve the documentation website.
-serve-debug: .conda/docs
-	$(CONDA) .conda/docs
-	mkdocs serve -a 127.0.0.1:8000 --verbose
+docs-serve-debug: env-docs
+	poetry run mkdocs serve -a 127.0.0.1:8000 --verbose
 
 # The --dirty flag makes mkdocs not regenerate everything when change is detected but also seems to
 # break references.
-serve-dirty: .conda/docs
-	$(CONDA) .conda/docs
-	mkdocs serve --dirty -a 127.0.0.1:8000
+docs-serve-dirty: env-docs
+	poetry run mkdocs serve --dirty -a 127.0.0.1:8000
 
-# Build the documentation website.
-site: .conda/docs $(shell find docs -type f) mkdocs.yml
-	$(CONDA) .conda/docs
-	mkdocs build --clean
-	rm -Rf site/overrides
 
-upload: .conda/docs $(shell find docs -type f) mkdocs.yml
-	$(CONDA) .conda/docs
-	mkdocs gh-deploy
+docs-upload: env-docs $(shell find docs -type f) mkdocs.yml
+	poetry run mkdocs gh-deploy
 
 # Check that links in the documentation are valid. Requires the lychee tool.
-linkcheck: site
+docs-linkcheck: site
 	lychee "site/**/*.html"
+
+# Start the trubot slack app.
+trubot:
+	poetry run python -u examples/trubot/trubot.py
+
+# Run a test with the optional flag set, meaning @optional_test decorated tests
+# are run.
```

Comment:
is this supposed to be describing `coverage`? It kinda does, but it's mostly about the coverage part isn't it?